### DISCUSSION TITLE
GraphSage on unseen nodes

### DISCUSSION THREAD

**Discussion by girwin1** [pyg-team/pytorch_geometric#9309] [Q&A]
I have trained a graphsage model on a graph. I now want to get the embeddings for new unseen nodes. 
So I pass to the model the node features and the edge index but then i get this error: 

features = torch.FloatTensor(feature_list)
edges = torch.Tensor([src, dst]).type(torch. int64)
model(features, edges)

IndexError: Dimension out of range (expected to be in range of [-1, 0], but got -2)

Any ideas?

Also, the node has edges to other unseen nodes. Should these edges be included in the data passed to the model? 

Thank you for any help!

**Comment by rusty1s** [comment]
What is the final shape of `features` and `edges`?

**Reply by girwin1** [reply]
Features is [9] as i have 1 node with 9 features and edges is [2,7]

**Reply by rusty1s** [reply]
How can you only have 1 node, but 7 edges? It looks like you are missing features of neighbors.

**Comment by rusty1s** [comment]
> Also, the node has edges to other unseen nodes. Should these edges be included in the data passed to the model?

Ideally, yes.
